pkg/transport/process/extension/stdio_executable.go

```diff
@@ -0,0 +1,69 @@
+package extension
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/gardener/component-cli/pkg/transport/process"
+)
+
+type stdIOExecutable struct {
+	processor *exec.Cmd
+	stdin     io.WriteCloser
+	stdout    io.Reader
+}
+
+// NewStdIOExecutable runs resource processor extension executable in the background.
+// It communicates with this processor via stdin/stdout pipes.
+func NewStdIOExecutable(ctx context.Context, bin string, args ...string) (process.ResourceStreamProcessor, error) {
+	cmd := exec.CommandContext(ctx, bin, args...)
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, err
+	}
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+	cmd.Stderr = os.Stderr
+
+	err = cmd.Start()
```

Comment:
shoudn't we start the binary when `Process` is called?
As wemight have the processor around for more time which means that the executable has to wait and listen to std.in